### PR TITLE
Significantly improve playlist file handling

### PR DESCRIPTION
**Pull request by rexendevar**
My git workflow is filthy so you'll have to veto a lot of these changes. Like a lot of them. Sorry. I'm submitting this with the intention of fixing it over time, as it is fully feature complete for everything I need it to be. 

New features:
- Updated playlist import/export dialog. 
- Playlists now default to a specific filepath rather than simply a containing folder. (If the path ends in a trailing slash it will be treated like a folder though.)
- When a playlist specifies a direct filepath, Tauon can also automatically IMPORT changes from this file, if its filesize changes. This occurs when the user switches to the playlist.
- Newly imported playlists default to auto-export, auto-import and relative file paths.
- New "default playlist export folder" option in Function settings. This determines the folder that playlists will export to by default.
- Optionally, this folder can also be automatically scanned for new and unimported playlist files. This scanning will only occur when the user runs a manual database rescan.

Limitations/known issues/possibilities:
- What if you update a playlist file that should auto-import, then close Tauon without switching to it? Only just thought of this, have to handle it
- When a playlist file is rescanned, it won't do anything with newly found radio stations. I don't plan to change this unless others consider it important
- Windows/Mac compatibility definitely broken

### CODE REVIEW

**Comment by Taiko2k** [comment]
The playlist import code really needs to be its own function, something like `pctl.check_update_playlist_from_file()` . Because we will want to run this elsewhere, maybe even on startup, not just switch playlist.

**Comment by rexendevar** [comment]
> Because we will want to run this elsewhere, maybe even on startup, not just switch playlist.

How about we only run it if "auto regenerate playlists" is enabled?

I realize that a lot of Tauon's design is geared toward doing a bunch of stuff at the very start and end and not really touching the disk during runtime, but in my testing the biggest slowdown I experienced was MAYBE a quarter second, and that's on a playlist of 500 songs. To me that seems like a small price to pay for what ends up being a pretty big feature.

I'm assuming you mean to take it out of switch playlist btw. if you mean we should have it in BOTH switch_playlist and on program load then I"m perfectly happy with that

**Comment by rexendevar** [comment]
> The playlist import code really needs to be its own function, something like `pctl.check_update_playlist_from_file()` . Because we will want to run this elsewhere, maybe even on startup, not just switch playlist.

done

**Comment by rexendevar** [comment]
new additions:
- switched to pathlib for everything important. didn't do everything because it seemed like a lot more trouble than it was worth but I'm 90% sure it should be fully OS compatible
- auto-reloaded playlists now import their radio stations
- invalid playlist folder path now has a visual indicator (renders in a different color) and resets itself if the user closes the menu without fixing it
- attempted error handling for relative paths. can't test if it works but it should
- relative paths explainer is localized to user's OS and library
- deprecated "show" parameter in playlist export functions as it was never used
- if user drags in a playlist file that's already in the library, tauon will now switch to it

File: src/tauon/t_modules/t_prefs.py [hunk 1]
```diff
@@ -322,6 +322,9 @@ class Prefs:
 	sync_playlist: int | None = None
 	download_playlist: int | None = None
 
+	autoscan_playlist_folder = False
```
Comment (C0rn3j) [inline src/tauon/t_modules/t_prefs.py]: While it may look obvious, it's best to define the type for these two new vars too (bool and str)

File: src/tauon/t_modules/t_main.py [hunk 2]
```diff
@@ -22207,36 +22388,47 @@ def __init__(self, tauon: Tauon) -> None:
...
 			if key not in ids:
 				del self.prefs.playlist_exports[key]
 
 	def render(self) -> None:
+		"""runs every frame that the playlist export menu is open.
+		also deals with the export entry logic."""
 		gui = self.gui
 		ddt = self.ddt
 		colours = self.colours
+		slash = "\\" if (self.tauon.system == "Windows" or self.tauon.msys) else "/"
```
Comment (C0rn3j) [inline src/tauon/t_modules/t_main.py:22423]: Is there a reason to support playlist files with no extension?

Because if we didn't, we can throw out the entire logic with leaving slashes in string paths and further processing them, and we won't have to rely on string paths.

We could simply say that anything without an extension is a directory.

Comment (rexendevar) [reply src/tauon/t_modules/t_main.py:22423]: I was originally planning for anything without an extension to be a directory but Taiko said directory should be only when it ends in a slash and if it doesn't end in either then it's a file

Comment (rexendevar) [reply src/tauon/t_modules/t_main.py:22423]: also it's not supporting playlist files with no extension, this part is literally just handling the text displaying in the path box. every playlist has a relevant extension when it's exported, but we can't generate it in the body of the render function because then it appears every frame and it's a pain in the ass to edit the text

File: src/tauon/t_modules/t_prefs.py (hunk 1 above)
Comment (rexendevar) [reply src/tauon/t_modules/t_prefs.py]: will do 👍

**Comment by rexendevar** [comment]
i wanna write the changelogs for this if that's alright btw

this will close #1612

**Comment by rexendevar** [comment]
i suspect a couple edge cases with relative paths and network tracks interacting but it's not something i can test to see how or if it actually fails btw
